### PR TITLE
[FIX] account_edi_ubl_cii: remove deprecated peppol eas

### PR DESCRIPTION
**Pull request by aliyatastemirova**
As of April 1, 2024, the migration period set by the Peppol Authority of Finland for the requirement specified in the Peppol Authority Specific Requirements document has ended. According to this requirement, Finnish end users’ Peppol addresses (participant identifiers) must adhere to the ISO 6523 code list 0216 OVT-format. Other address types are not allowed for Finnish end users.

See also:
[Finland Peppol Authority requirements](https://peppol.org/wp-content/uploads/2023/08/Finland-Peppol-Authority-Specific-Requirements.pdf)
[Finland Peppol Authority website](https://www.valtiokonttori.fi/en/service/the-state-treasury-is-the-finnish-peppol-authority/#for-service-providers_authority-specific-requirements-in-finland)

It turned out that we have a few others that are no longer used on Peppol.
We will remove those in master as well.
See:
[Peppol codelists](https://docs.peppol.eu/edelivery/codelists/)

no task, reported by the Finnish Peppol Authoirty Team

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by aliyatastemirova** [comment]
Peppol participants, who use any of these EAS codes, are being deregistered on the IAP

**Comment by aliyatastemirova** [comment]
@fw-bot up to saas-17.2

File: addons/account_edi_ubl_cii/models/res_partner.py [hunk 1]
```diff
@@ -125,6 +125,16 @@ class ResPartner(models.Model):
         ]
     )
 
+    @api.constrains('peppol_eas')
+    def _check_peppol_eas(self):
+        for partner in self:
+            if partner.peppol_eas in ('0212', '0213'):
```
Comment (JulienVR) [inline addons/account_edi_ubl_cii/models/res_partner.py:131]: funny how this is not mentioned in https://docs.peppol.eu/edelivery/codelists/changelog.html but it is by the Finish Authority :sweat:, so we can just rely on the docs.peppol.eu website...

**Review by JulienVR — APPROVED** [review]
Thank you for the work ! :100: 

As discussed, since users can register from saas-16.4, let's prevent them from doing so with an invalid EAS adding something like `self.company_id.partner_id._check_peppol_eas()` inside `button_create_peppol_proxy_user`. I'll let you do that during the fw-port.
 
@robodoo r+

**Pull request closed without merging**